airflow/models/taskinstance.py

```diff
@@ -3615,7 +3615,10 @@ def duration_expression_update(
             return query.values(
                 {
                     "end_date": end_date,
-                    "duration": (func.julianday(end_date) - func.julianday(cls.start_date)) * 86400,
+                    "duration": func.cast(
```

Comment:
Another alternative is:

```sql
sqlite> SELECT ROUND((julianday('2024-06-01T01:00:00') - julianday('2024-06-01T00:00:00')) * 86400, 0);
3600.0
```

Reply:
Or we could simplify and just do the following since we just care about seconds:

```sql
sqlite> SELECT strftime('%s', '2024-06-01T01:00:00') - strftime('%s', '2024-06-01T00:00:00');
3600
```

thoughts @ashb ?

Reply:
Option 3:

```sql
sqlite> SELECT unixepoch('2024-06-01T01:00:00.1234', 'subsec') - unixepoch('2024-06-01T00:00:00', 'subsec');
3600.12299990654
```

Reply:
oh nice, I like that

Reply:
Well, problem! :(

- `unixepoch` was added in Sqlite 3.38.0 [`2022-02-22 (3.38.0)
`](https://www.sqlite.org/changes.html#:~:text=Added%20the%20unixepoch()%20function.)
- `subsec` was added in Sqlite 3.42.0 [`2023-05-16 (3.42.0)`](https://www.sqlite.org/changes.html#:~:text=Added%20the%20subsecond%20modifier%20to%20the%20date%20and%20time%20functions.)

Reply:
and we support Sqlite 3.15.0+

Reply:
I am planning to go with:

```sql
sqlite> SELECT
   ...> (strftime('%s', '2024-06-01T01:00:00.456') - strftime('%s', '2024-06-01T00:00:00.123'))
   ...> + ROUND((strftime('%f', '2024-06-01T01:00:00.456') - strftime('%f', '2024-06-01T00:00:00.123')), 3);
3600.333
```

Reply:
Updated: https://github.com/apache/airflow/pull/45002/commits/08f56c320ae0db2ccfbb1a7ec9a95dc1be9eae01

Reply:
One more familiar source: https://til.simonwillison.net/sqlite/unix-timestamp-milliseconds-sqlite

Reply:
We maintain 6 decimal places for TI table. Do we want to maintain something similar? 
I am ok with either cases.

Reply:
Another option is to just work with format:
```
SELECT printf('%.6f', (julianday('2024-06-01T01:00:00.123456') - julianday('2024-06-01T00:00:00')) * 86400);
```

Which does a format and hence no rounding or any logic involved, it just truncates the output, consistently

Reply:
@amoghrajesh You will get the same problem with that:

```
sqlite> SELECT printf('%.6f', (julianday('2024-06-01T01:00:00') - julianday('2024-06-01T00:00:00')) * 86400);
3599.999987
```

Reply:
I am ok with the current approach.